src/app/components/client/PremiumBadge.tsx

```diff
@@ -56,6 +57,18 @@ export default function PremiumBadge({
     dialogState,
   );
 
+  const { update } = useSession();
+
+  useEffect(() => {
+    async function updateSession() {
+      await update();
+    }
+    void updateSession();
+
+    // This should only run once per page load - `update` will always appear to be changed.
```

Comment:
We have two places where we are adding the `<PremiumBadge>` — `MobileShell` and `Toolbar`. I’m wondering if we should create a component (e.g. `<PremiumCheck>`) that is only returning a Fragment `<></>` that is doing the check and adding that to `Shell.tsx`?

Reply:
We actually do have that already https://github.com/mozilla/blurts-server/blob/f0e17f84611c0d0a0758c54891c3facac8b644ab/src/app/components/client/SubscriptionCheck.tsx#L11 - I can change to using this approach instead.

Reply:
Hm OK I gave this a try but test failures started showing up in other tests, and I don't know enough about the test setup to resolve this... I'm going to land as-is because I believe it improves the situation, but moving this out of `PremiumBadge` and into `Shell` directly is worth doing I believe, I'll file a follow-up issue.

Reply:
Filed https://mozilla-hub.atlassian.net/browse/MNTOR-2396